draft-protected-headers.md

```diff
@@ -135,9 +135,99 @@ Due to compatibility and usability concerns, a Mail User Agent SHOULD NOT obscur
+ - `From`
+ - `To`
+ - `Cc`
+ - `Date`
+
+The above is a complete list.  No other headers are considered "typically visible".
+
+Other headers may affect the visible rendering of the message (e.g., `References` and `In-Reply-To` may affect the placement of a message in a threaded discussion), but they are not directly displayed to the user and so are not considered "typically visible" for the purposes of this document.
+
+Message Generation: Including a Legacy Display Part
+---------------------------------------------------
+
+A generating MUA that wants to make an obscured Subject (or any other typically visible header) visible to a recipient using a legacy MUA SHOULD modify the Cryptographic Payload by wrapping the intended body of the message in a `multipart/mixed` MIME part that prefixes the intended body with a Legacy Display part.
+
+The Legacy Display part MUST be of Content-Type `text/rfc822-headers`, and MUST contain a `protected-headers` parameter whose value is `v1`.
+It SHOULD be marked with `Content-Disposition: inline` to encourage recipients to render it.
+
+The conents of the Legacy Display part MUST be only the typically visible headers that the sending MUA intends to obscure after encryption.
+
+The original body (now a subpart) SHOULD also be marked with `Content-Disposition: inline` to discourage legacy clients from presenting it as an attachment.
+
+### Legacy Display Transformation {#legacy-display-transformation}
+
+Consider a message whose Cryptographic Payload, before encrypting, that would have a traditional `multipart/alternative` structure:
+
+    X └┬╴multipart/alternative
+    Y  ├─╴text/plain
+    Z  └─╴text/html
+
+When adding a Legacy Display part, this structure becomes:
+
+    V └┬╴multipart/mixed
+    W  ├─╴text/rfc-822-headers ("Legacy Display" part)
+    X  └┬╴multipart/alternative ("original body")
+    Y   ├─╴text/plain
+    Z   └─╴text/html
+
+Note that with the inclusion of the Legacy Display part, the Cryptographic Payload is the `multipart/mixed` part (part `V` in the example above), so Protected Headers should be placed at that part.
+
+### When to Generate Legacy Display
+
+A MUA SHOULD transform a Cryptographic Payload to include a Legacy Display part only when:
+
+ - The message is going to be encrypted, and
+ - At least one typically visible header (see {{#typically-visible-header}}) is going to be obscured
+
+Additionally, if the sender knows that the recipient's MUA is capable of interpreting Protected Headers, it SHOULD NOT attempt to include a Legacy Display part.
+(Signalling such a capability is out of scope for this document)
+
+Message Rendering: Omitting a Legacy Display Part
+-------------------------------------------------
+
+A MUA that understands Protected Headers may receive an encrypted message that contains a Legacy Display part.
+Such an MUA SHOULD avoid rendering the Legacy Display part to the user at all, since it knows and can render the actual Protected Headers.
```

Comment:
typo: knows how ?

Reply:
... knows how to?  Umm. :stuck_out_tongue_winking_eye:

Reply:
it knows the actual protected headers.

maybe "is aware of" instead of "knows" ?